### PR TITLE
Allow to use api for different Freshdesk accounts

### PR DESCRIPTION
**Pull request by artem-boltunov-h**

### CODE REVIEW

File: lib/freshdesk-rest.rb [hunk 1]
```diff
@@ -21,5 +22,12 @@ def self.reset
     def self.configure
       yield(configuration)
     end
+
+    class << self
+      extend Gem::Deprecate
+      deprecate :configure, :none, 2024, 5
+      deprecate :configuration, :none, 2024, 5
+      deprecate :reset, :none, 2024, 5
```
Comment (diaclavijo) [inline lib/freshdesk-rest.rb:30]: i learnt about this now

File: lib/freshdesk-rest/configuration.rb [hunk 2]
```diff
@@ -17,6 +17,11 @@ def domain
         raise 'Freshdesk domain not defined' if @domain.nil?
         @domain
       end
+
+      extend Gem::Deprecate
+      deprecate :initialize, :none, 2024, 5
+      deprecate :api_key, :none, 2024, 5
+      deprecate :domain, :none, 2024, 5
```
Comment (diaclavijo) [inline lib/freshdesk-rest/configuration.rb:24]: Why? we can still keep this and make it to be used by default in `Freshdesk::Rest::Client`

Comment (artem-boltunov-h) [reply lib/freshdesk-rest/configuration.rb]: It makes total sense, thank you! I rebased the PR and added default configuration option to the client.

**Review by diaclavijo — APPROVED** [review]
wonderful!

**Review by d4rky-pl — APPROVED** [review]

**Pull request merged**